### ISSUE TITLE
Inaccessible pods on other nodes for high availability cluster

### ISSUE THREAD

**Issue by Gentatsu** [ubuntu/microk8s#1892]
I made a 3 node cluster on ec2, and wanted to just launch a generic application to make sure everything is accessible. I created a microk8s environment on each machine, and got them to add to a HA cluster. When I tried to launch the `microbot` deployment on ubuntu tutorial, each machine could only access its own pod. 

When running `microk8s kubectl get pods -o wide`, I get the following:

```
NAME                        READY   STATUS    RESTARTS   AGE    IP            NODE               NOMINATED NODE   READINESS GATES
microbot-5f5499d479-ngz56   1/1     Running   0          179m   10.1.94.76    ip-172-31-18-128   <none>           <none>
microbot-5f5499d479-nkctv   1/1     Running   1          175m   10.1.162.72   ip-172-31-21-37    <none>           <none>
microbot-5f5499d479-zkjcn   1/1     Running   1          175m   10.1.162.73   ip-172-31-21-37    <none>           <none>
```
This is with the deployment scaled to 3. 

If I curl on the machine ending in 128, I get a 1/3 chance in hitting itself, and 2/3 with the one ending in 37. On my other machine, it always hangs because it is attempting to access one of the other two machines and does not seem to be able to. 

ufw is disabled, and I've tried running 

```
sudo iptables -P FORWARD ACCEPT
sudo apt-get install iptables-persistent
   ```

on each machine, to no avail. I can ping them on other services fine. I enabled ingress w/o a service, and each one shoots me a 404 error, so it can clearly route. 

I've attached the inspection logs.
[inspection-report-20210112_132731.tar.gz](https://github.com/ubuntu/microk8s/files/5802580/inspection-report-20210112_132731.tar.gz)

**Comment by ktsakalozos** [comment]
This looks like a node to node communication problem. Not sure what the problem is right now. Could you temporarily make the inbound and outbound security rules permissive to quickly test that AWS is not getting in the way?

**Comment by Gentatsu** [comment]
I went overkill and opened every port inbound/outbound juuuuust to make sure nothing was getting in the way. I did this originally because the docs weren't clear about which ports are required to be open for adding nodes.

**Comment by rockaut** [comment]
I also had problems on my Ubuntu Cluster, for me it turned out to be problems with net_bridge. So I had to enable the modules and sysctl.

Added to `/etc/modules-load.d/modules.conf`:

```text
overlay
br_netfilter
bridge
```

and to `/etc/sysctl.conf`:

```text
net.bridge.bridge-nf-call-ip6tables = 1
net.ipv4.ip_forward = 1
net.bridge.bridge-nf-call-iptables = 1
```

Rebooted and all worked.

I previously had docker already installed and uninstalled and also played around with cni and podman.

**Comment by rockaut** [comment]
I dropped a comment over at the forum to maybe also leave mention it in the docs: https://discuss.kubernetes.io/t/troubleshooting/11267/2

**Comment by Gentatsu** [comment]
Afraid I'm still having the same issue :/ even with those changes to all my servers.

**Comment by Gentatsu** [comment]
Scratch that. I locked down my inbound rules after this didn't work until I revisited it today. 

I opened up all inbound traffic again, and it seems to work. I can curl localhost + either ip and get routed to a pod. 

Is there a list of ports required to be open for this to work node to node internally? I opened all the ones on here: https://microk8s.io/docs/ports but it doesn't work with them open. I've made them all TCP, so I'm unsure if any need to be UDP or if there are any ports not mentioned required for comms?
